meson.build

```
@@ -1701,6 +1701,13 @@ features += {'libmpv-' + get_option('default_library'): get_option('libmpv')}
 
 # build targets
 if win32
+    # Older meson versions generate this in the player subdirectory.
+    if get_option('cplayer') and meson.version().version_compare('>= 1.3.0')
+        wrapper_sources= 'osdep/win32-console-wrapper.c'
+        executable('mpv', wrapper_sources, c_args: '-municode', link_args: '-municode',
+                   name_suffix: 'com', install: true)
+    endif
+
```

There is a way to make the old (current) version of meson generate the .com in the same place as the .exe (and get rid of `player/meson.build` - just don't forget to add `subdir('player/javascript')` and `subdir('player/lua')`). 
```suggestion
    wrapper_sources= 'osdep/win32-console-wrapper.c'
    if get_option('cplayer') and meson.version().version_compare('>= 1.3.0')     
        executable('mpv', wrapper_sources, c_args: '-municode', link_args: '-municode',
                   name_suffix: 'com', install: true)
    else
        executable('mpv.com', wrapper_sources, c_args: '-municode', link_args: '-municode',
                   name_suffix: '.', install: true)
    endif
```

---

Won't that just create `mpv.com..`?

---

No, it creates `mpv.com`, I've tested it.

---

Is that windows specific behavior or something? It doesn't here on a test case which is what I'd expect.

---

I've tested it on MSYS2. Its build directory is `mpv.com...p`, but the file is `mpv.com`.

---

That seems bizarre. It must be windows itself not allowing the filename to end in `..` like what should be happening as indicated by the `mpv.com...p`. I guess it might technically work, but I don't think we need to rely on suck hacks.